docs/JAX_basic_text_classification.md

```diff
@@ -0,0 +1,455 @@
+    )
+```
+
+```{code-cell} ipython3
+eval_metrics = nnx.MultiMetric(
+    loss=nnx.metrics.Average('loss'),
+    accuracy=nnx.metrics.Accuracy(),
+)
+
+
+train_metrics_history = {
+    "train_loss": [],
+}
+
+eval_metrics_history = {
+    "test_loss": [],
+    "test_accuracy": [],
+}
+```
+
+```{code-cell} ipython3
+import tqdm
+
+
+bar_format = "{desc}[{n_fmt}/{total_fmt}]{postfix} [{elapsed}<{remaining}]"
+train_total_steps = len(train_dataset) // train_batch_size
+
+
+def train_one_epoch(epoch):
+    model.train()  # Set model to the training mode: e.g. update batch statistics
+    with tqdm.tqdm(
+        desc=f"[train] epoch: {epoch}/{num_epochs}, ",
+        total=train_total_steps,
+        bar_format=bar_format,
+        leave=True,
+    ) as pbar:
+        for batch in train_loader:
+            loss = train_step(model, optimizer, batch)
+            train_metrics_history["train_loss"].append(loss.item())
+            pbar.set_postfix({"loss": loss.item()})
+            pbar.update(1)
+
+
+def evaluate_model(epoch):
+    # Compute the metrics on the train and val sets after each training epoch.
+    model.eval()  # Set model to evaluation model: e.g. use stored batch statistics
+
+    eval_metrics.reset()  # Reset the eval metrics
+    for test_batch in test_loader:
+        eval_step(model, test_batch, eval_metrics)
+
+    for metric, value in eval_metrics.compute().items():
+        eval_metrics_history[f'test_{metric}'].append(value)
+
+    print(f"[test] epoch: {epoch + 1}/{num_epochs}")
+    print(f"- total loss: {eval_metrics_history['test_loss'][-1]:0.4f}")
+    print(f"- Accuracy: {eval_metrics_history['test_accuracy'][-1]:0.4f}")
+```
+
+Let us now start the training
+
+```{code-cell} ipython3
+for epoch in range(num_epochs):
+    train_one_epoch(epoch)
+    evaluate_model(epoch)
+```
+
+Let us visualize collected metrics
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+
+
+plt.plot(train_metrics_history["train_loss"], label="Loss value during the training")
+plt.legend()
+```
+
+```{code-cell} ipython3
+fig, axs = plt.subplots(1, 2, figsize=(10, 10))
+axs[0].set_title("Loss value on test set")
+axs[0].plot(eval_metrics_history["test_loss"])
+axs[1].set_title("Accuracy on test set")
+axs[1].plot(eval_metrics_history["test_accuracy"])
+```
+
+We can observe that the model starts overfitting after the 5-th epoch and the best accuracy it could achieve is around 0.87. Let us also check model's predictions on the test data:
+
+```{code-cell} ipython3
+data = test_dataset[10]
+```
+
+```{code-cell} ipython3
+text_processing = TextPreprocessing(tokenizer, max_length=max_length)
+processed_data = text_processing.map(data)
+```
+
+```{code-cell} ipython3
+model.eval()
+preds = model(processed_data["text"][None, :])
+preds, preds.argmax(axis=-1)
+```
+
+```{code-cell} ipython3
+print("Text:\n", data["text"])
+print("Expected label:", data["label"])
+print("Predicted label:", preds.argmax(axis=-1).item())
+```
+
```

Comment:
Is it possible to include a quick summary/interpretation of what we are seeing in these labels?